### PR TITLE
Added a count function for lists

### PR DESCRIPTION
**Pull request by etorreborre**
I've been writing this a few times now so it's time to put it in `p` I guess.

### CODE REVIEW

File: test/Test/P/List.hs [hunk 1]
```diff
@@ -19,6 +19,10 @@ prop_lastMaybe_empty :: Property
 prop_lastMaybe_empty =
   lastMaybe [] === (Nothing :: Maybe ())
 
+prop_count :: [Int] -> Property
+prop_count list =
+  count (>= 0) list === length (filter (>= 0) list) 
+
```
Comment (domdere) [inline test/Test/P/List.hs]: I think more generic properties for this function should include checks on the output for `(===) <$> count (const True) <*> id` and `count (const False)`.

Comment (domdere) [inline test/Test/P/List.hs]: Also should be able to get some `a -> Bool` functions from `CoArbitrary` so should be able to let it take a generated predicate?

**Comment by domdere** [comment]
:+1:

File: test/Test/P/List.hs (hunk 1 above)
Comment (domdere) [inline test/Test/P/List.hs]: Sorry, the inability to adequately `show` the `a -> Bool` function hurts the reproducibility of any failures, my bad, its probably not worth it @etorreborre

Comment (EduardSergeev) [inline test/Test/P/List.hs]: With `Test.QuickCheck.Function` it is just
```haskell
prop_count :: [Int] -> (Fun Int Bool) -> Property
prop_count list (Fun _ f) =
  count f list === length (filter f list)
```
And it does produce something similar to
```
*** Failed! Falsifiable (after 1 test and 2 shrinks):  
[]
{_->False}
1 /= 0
```
Looks useful to me

Comment (domdere) [inline test/Test/P/List.hs]: oh cool yeh that is better, does also produce less trivial functions?

Comment (etorreborre) [inline test/Test/P/List.hs]: Thanks @EduardSergeev that's what I was looking for.

**Pull request merged**